plutus-core/plutus-ir/src/PlutusIR/Transform/Inline/CallSiteInline.hs

```diff
@@ -154,7 +105,11 @@ fullyApplyAndBetaReduce info args0 = do
               (\n -> if n == param then Just <$> PLC.rename arg else pure Nothing)
               acc
           go acc' arity' args'
-        _ -> pure Nothing
+        -- term/type argument mismatch, don't inline
+        (TermParam _:_, TypeAppContext{}) -> pure Nothing
+        (TypeParam _:_, TermAppContext{}) -> pure Nothing
```

Comment:
I’m thinking when we have a mismatch of term/type arguments, we don’t inline. This shouldn’t happen and so we leave it alone. Do people agree? Like if we do inline in such case for the correctly applied earlier arguments, it may reduce size, but I feel there’s a risk of making things more wrong?